### PR TITLE
Add support for the LXA Optick latency measurement util

### PR DESCRIPTION
**Pull request by hnez**
The LXA Optick board is an utility for measuring latencies, that supports both measuring the latencies involved in realtime Linux
and microcontroller applications (on the order of nanoseconds), as well as light source -> camera -> computer -> display -> light sensor round trip latencies.

(Hence the name, Op-Tick, like being something visual but also clock like, you know ... a play on words ... nevermind).

The Optick board comes with support hardware for these optical latency measurements.

The board has two event-generating output ports and two event-capturing input ports, that capture and generate events relative to a local 64 bit counter running at 48MHz.
The counters from different Optick devices can be synchronized using the PPS output of a GPS module, to support e.g. time sensitive networking latency measurements over some distance.

The inputs are analog and feature an on-board comparator that compares the input voltage to a software configurable reference voltage to form a basic input trigger.

This commit also contains a commandline utility that implements the basic optick usecase of measuring a glass to glass camera / display latency, by first determining the correct trigger level, generating and capturing events and lastly generating a histogram of measured latencies.

TODO before merging:

  - [x] This PR is based on #38, which is in turn based on #37, which is based on #36, so all of these should be merged before this one.
  - [x] Do some more manual testing with the Optick device connected to an IOBus server and the commandline util.
  - [x] This PR is based on #40, which should be merged first.
  - [x] Update the firmware blob to a released one.

### CODE REVIEW

**Comment by SmithChart** [comment]
#38 has been merged. Please rebase.

**Comment by hnez** [comment]
Great to see that we are coming along with merging the PRs so nicely!

I've rebased this PR on top of recent master, but have also converted it to a draft PR, since it should not be merged before we have a released version of the Optick Firmware to include in the firmware folder.

I've created #40, which breaks out the first two commits of this PR so we are not held back too much by the firmware blob.

File: lxa_iobus/cli/optick.py [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+
+    if mid > 0.95:
+        print("WARNING: The trigger level is very high.")
+        print("Is the sensor protected from light sources like e.g. direct sunlight?")
+        print()
+
+    if low_correct < 0.95 or high_correct < 0.95:
+        print("WARNING: The trigger level has bad separation")
+        print("Are all connections good, no stray light entering the setup")
+        print("and automatic brightness adjustments disabled as much as possuble?")
```
Comment (hnez) [inline lxa_iobus/cli/optick.py:125]: Reviewing this part may be a good fit for @ukleinek, who has a knack for math stuff.

**Comment by hnez** [comment]
Alright, we now have an Optick firmware image and are ready for review! 🎉

**Review by SmithChart — APPROVED** [review]

File: lxa_iobus/node/products.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ class Iobus4Do3Di3Ai(Node):
 
     NAME_PREFIX = "4DO-3DI-3AI-00005."
     FIRMWARE_FILE = "lxatac_can_io-t01.bin"
-    FIRMWARE_VERSION = (0, 6, 0)
+    FIRMWARE_VERSION = (0, 5, 0)
```
[the remark this answers is not in the document]
Comment (SmithChart) [reply lxa_iobus/node/products.py]: This change is part of and old version of this PR. Please ignore.

File: lxa_iobus/cli/optick.py (hunk 1 above)
Comment (hnez) [reply lxa_iobus/cli/optick.py:125]: Yes, sure. I am not sure if the percent values I print here are statistically sound, but they have been helpful enough, so I think we are good here.

**Pull request merged**